web3/eth.py

```diff
@@ -53,91 +75,121 @@
 from web3.module import (
     Module,
 )
+from web3.types import (
+    ENS,
+    BlockData,
+    BlockIdentifier,
+    FilterParams,
+    GasPriceStrategy,
+    LogParams,
+    MerkleProof,
+    SyncStatus,
+    TxParams,
+    TxReceipt,
+    Uncle,
+    Wei,
+)
 
 
 class Eth(Module):
     account = Account()
     defaultAccount = empty
     defaultBlock = "latest"
-    defaultContractFactory = Contract
+    defaultContractFactory: Type[Union[Contract, ConciseContract, ContractCaller]] = Contract  # noqa: E704,E501
     iban = Iban
     gasPriceStrategy = None
 
-    def namereg(self):
+    def namereg(self) -> NoReturn:
         raise NotImplementedError()
```

Comment:
@kclowes these APIs can be removed.  They won't ever be implemented and they aren't needed.  Just carryovers from a very long time ago.